cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1179,31 +1249,21 @@ public RunRecordDetail recordProgramRejected(ProgramRunId programRunId,
       Map<String, String> runtimeArgs, Map<String, String> systemArgs,
       byte[] sourceId, @Nullable ArtifactId artifactId)
       throws IOException {
-    long startTs = RunIds.getTime(programRunId.getRun(), TimeUnit.SECONDS);
-    if (startTs == -1L) {
-      LOG.error(
-          "Ignoring unexpected request to record provisioning state for program run {} that does not have "
-
-              + "a timestamp in the run id.", programRunId);
-      return null;
-    }
-
     RunRecordDetail existing = getRun(programRunId);
-    // for some reason, there is an existing run record?
-    if (existing != null) {
+
+    // Return null if for some reason the run record doesn't exist.
+    if (existing == null) {
       LOG.error(
-          "Ignoring unexpected request to record rejected state for program run {} that has an existing "
-              + "run record in run state {} and cluster state {}.",
-          programRunId, existing.getStatus(), existing.getCluster().getStatus());
+          "Ignoring unexpected request to record rejected state for program run {} that has no existing run record.",
```

Comment:
Why? I don't know in which case it may happen, but I'd still leave a trace. Just skip `delete`.

Reply:
Right.

Not returning early and skipping delete if run record doesn't exist.